test/node/tools/utils.js

```diff
@@ -125,3 +125,33 @@ const bufferFromHexArray = array => {
 };
 
 exports.bufferFromHexArray = bufferFromHexArray;
+
+/**
+ * A helper to calculate the byte size of a string (including null)
+ *
+ * ```js
+ * const x = stringToUTF8HexBytes('ab') // { x: '03000000616200' }
+ *
+ * @param string - representing what you want to encode into BSON
+ * @returns BSON string with byte size encoded
+ */
+const stringToUTF8HexBytes = str => {
+  var b = Buffer.from(str, 'utf8');
+  var len = b.byteLength;
+  var out = Buffer.alloc(len + 4 + 1);
+  out.writeInt32LE(len + 1, 0);
+  out.set(b, 4);
+  return out.toString('hex');
```

Comment:
```suggestion
  out[len+1] = 0x00;
  return out.toString('hex');
```

The buffer is zero-ed out so this isn't exactly necessary but just to be sure the null byte is there at the end seems like a good idea.